tests/spec/create.spec.js

```diff
@@ -17,52 +17,97 @@
-const { superspawn } = require('cordova-common');
+const os = require('os');
+const path = require('path');
+const xcode = require('xcode');
+const create = require('../../bin/lib/create');
 
-const cordova_bin = path.join(spec, '../..', 'bin');
-const tmp = require('tmp').dirSync().name;
+const makeTempDir = () => path.join(
+    fs.realpathSync(os.tmpdir()),
+    `cordova-ios-create-test-${Date.now()}`
+);
 
-function createAndBuild (projectname, projectid) {
-    const projectTempDir = path.join(`${tmp}/${projectname}`);
-    const createBin = path.join(`${cordova_bin}/create`);
-    const buildBin = path.join(`${projectTempDir}/cordova/build`);
+/**
+ * Verifies that some of the project file exists. Not all will be tested.
+ * E.g. App's resource directory, xcodeproj, xcworkspace, and CordovaLib.
+ *
+ * @param {String} tmpDir
+ * @param {String} projectName
+ */
+function verifyProjectFiles (tmpDir, projectName) {
+    expect(fs.existsSync(path.join(tmpDir, projectName))).toBe(true);
+    expect(fs.existsSync(path.join(tmpDir, `${projectName}.xcodeproj`))).toBe(true);
+    expect(fs.existsSync(path.join(tmpDir, `${projectName}.xcworkspace`))).toBe(true);
+    expect(fs.existsSync(path.join(tmpDir, 'CordovaLib'))).toBe(true);
+}
 
-    // Remove any pre-existing temp projects
-    fs.removeSync(projectTempDir);
+/**
+ * Verifies that the set bundle id matches with the expected.
+ *
+ * @param {String} tmpDir
+ * @param {String} projectName
+ * @param {String} expectedBundleIdentifier
+ */
+function verifyProjectBundleIdentifier (tmpDir, projectName, expectedBundleIdentifier) {
+    const pbxproj = path.join(tmpDir, `${projectName}.xcodeproj/project.pbxproj`);
+    const xcodeproj = xcode.project(pbxproj);
+    xcodeproj.parseSync();
+    const actualBundleIdentifier = xcodeproj.getBuildProperty('PRODUCT_BUNDLE_IDENTIFIER');
+    expect(actualBundleIdentifier).toBe(`"${expectedBundleIdentifier}"`);
+}
 
-    return superspawn.spawn(createBin, [projectTempDir, projectid, projectname], { printCommand: true }).then(
-        () => {
-            expect(true).toBe(true); // It is expected that create is successful
+/**
+ * Runs and expects for a successful build.
+ *
+ * @param {String} tmpDir
+ * @returns {Promise}
+ */
+function verifyBuild (tmpDir) {
+    const Api = require(path.join(tmpDir, 'cordova/Api.js'));
 
-            return superspawn.spawn(buildBin, ['--emulator'], { printCommand: true }).then(
-                () => {
-                    expect(true).toBe(true); // It is expected that build is successful
-                },
-                () => fail('Project Build has failed and is not expected.')
-            );
-        },
-        () => fail('Project create has failed and is not expected.')
-    ).finally(() => {
-        // Delete Temp Project
-        fs.removeSync(projectTempDir);
-    });
+    return new Api()
+        .build({ emulator: true })
+        .then(() => {
+            expect(true).toBe(true);
+        });
```

Comment:
How about
```suggestion
    return expectAsync(new Api().build({ emulator: true }))
        .toBeResolved();
```

Reply:
Yes, this will work. I should read up on Jasmine's APIs. I still have not read it.